### DISCUSSION TITLE
custom blog ejs template with N minute read function?

### DISCUSSION THREAD

**Discussion by MayaGans** [quarto-dev/quarto-cli#1797] [Q&A] [answered]
Hi everyone! I'm making my own custom blog page layout and I want to add a time estimate to read the article to my thumbnail. Here's what my YAML looks like:

```
listing:
  contents: posts
  sort: "date desc"
  template: templates/bloghome.ejs
page-layout: full
```

And here's a div soup of `bloghome.ejs` where I can see the 

- `item.title` blogpost title
- `item.path` link out to the actual post
- `item.image` a little cover photo
- `item.categories` an array of categories for the post
- `item.date` when the post was written 

**but I'd like to add to the thumbnail an "N minute read"** and i was kind of lazily hoping there's something that exists already that I can piggy back off of? Is there some metadata object I have access to here maybe? For instance I didn't specify `item.path` in my post YAML, I just got that for free - what, if anything else do I get for free? 

```{=html}
<div>
<% for (const item of items) { %>
  <a href="<%- item.path %>">
  <div class="post">
    <h3><%= item.title %></h3>
    <img src="<%- `${item.image}` %>" />
    <div class="date">Last Updated on <%- item.date %></div>
    <div class="date"><%
      /* 
        some function that will read in the file at item.path
        and then calculate the readtime using some JS magic maybe?
      */
    %></div>
    <div class="categories">
        <% for (let step = 0; step < item.categories.length; step++) { %>
          <div><%- item.categories[step] %></div>
        <% } %>
    </div>
  </div>
  </a>
<% } %>
</div>
```

I was thinking I can maybe use a function like this but im not sure how to grab the file if i need to do this from scratch. Anyone have any ideas? 

```
function readingTime() {
  const text = document.getElementById("article").innerText; // not sure how to grab the file... I know I have item.path.... can I use that?
  const wpm = 225;
  const words = text.trim().split(/\s+/).length;
  const time = Math.ceil(words / wpm);
  return time
}
readingTime();
```

Was also thinking this could be a cool feature to just add to posts/blog template if this is of interest to others!

**Comment by dragonstyle** [comment, marked as answer]
I believe that you could use `item['reading-time']` to reference the reading time... im afk but give that a try!

**Reply by MayaGans** [reply]
OMG thank you! Can you point me to the documentation or file where all these goodies are located? I wanna see what other fun metadata I can grab from a post!

**Reply by jjallaire** [reply]
It's right here: https://quarto.org/docs/websites/website-listings.html#listing-fields
